bindata/network/ovn-kubernetes-ipsec/ovn-ipsec.yaml

```diff
@@ -0,0 +1,207 @@
+            # Stop ovn-monitor-ipsec
+            /usr/share/openvswitch/scripts/ovs-ctl stop-ovs-ipsec
+            # Stop libreswan
+            /usr/libexec/ipsec/whack --shutdown
+            /sbin/ip xfrm policy flush
+            /sbin/ip xfrm state flush
+            /usr/sbin/ipsec --stopnflog
+            exit 0
+          }
+          trap cleanup SIGTERM
+
+          # Every time we restart the container, we will create a new key pair
+          echo "Configuring IPsec keys"
+
+          # We use the system-id as the CN. This is a requirement for OVN
+          cn=$(ovs-vsctl --retry -t 60 get Open_vSwitch . external-ids:system-id | tr -d "\"")
+
+          mkdir -p /etc/openvswitch/keys
+          pushd /etc/openvswitch/keys
+          umask 077 && openssl genrsa -out ipsec-privkey.pem 2048
+          openssl req -new -text \
+                  -extensions v3_req \
+                  -addext "subjectAltName = DNS:${cn}" \
+                  -subj "/C=US/O=ovnkubernetes/OU=kind/CN=${cn}" \
+                  -key ipsec-privkey.pem \
+                  -out ipsec-req.pem
+
+          csr_64=$(cat ipsec-req.pem | base64 | tr -d "\n")
+
+          # The signer controller does not allow re-signing a key. We will
+          # delete the old key to be sure it is not there
+          kubectl delete --ignore-not-found=true csr/$(hostname)
+
+          cat <<EOF | kubectl apply -f -
+          apiVersion: certificates.k8s.io/v1
+          kind: CertificateSigningRequest
+          metadata:
+            name: $(hostname)
+          spec:
+            request: ${csr_64}
+            signerName: network.openshift.io/signer
+            usages:
+            - ipsec tunnel
+          EOF
+
+          until [ ! -z $(kubectl get csr/$(hostname) -o jsonpath='{.status.certificate}' 2>/dev/null) ]
+          do
+            ((counter++))
+            if [ $counter -gt 60 ];
+            then
+                    echo "Unable to sign certificate after $counter seconds"
+                    exit 1
+            fi
+          done
+
+          kubectl get csr/$(hostname) -o jsonpath='{.status.certificate}' | base64 -d | openssl x509 -outform pem -text -out ipsec-cert.pem
+
+          kubectl delete csr/$(hostname)
+
+          cat /signer-ca/tls.crt | openssl x509 -outform pem -text > /etc/openvswitch/keys/ipsec-cacert.pem
+
+          ovs-vsctl --retry -t 60 set Open_vSwitch . other_config:certificate=/etc/openvswitch/keys/ipsec-cert.pem \
+                                                     other_config:private_key=/etc/openvswitch/keys/ipsec-privkey.pem \
+                                                     other_config:ca_cert=/etc/openvswitch/keys/ipsec-cacert.pem
+
+          popd
+
+          # Workaround for https://github.com/libreswan/libreswan/issues/373
+          ulimit -n 1024
+
+          /usr/libexec/ipsec/addconn --config /etc/ipsec.conf --checkconfig
+          # Check for kernel modules
+          /usr/libexec/ipsec/_stackmanager start
```

Comment:
do we need to bind-mount `/lib/modules` ?

Reply:
I think xfrm is built-in in this case as it seems to work and I dont see any xfrm modules when i do an `lsmod`